x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.test.tsx

```diff
@@ -172,6 +164,35 @@ describe('Credentials', () => {
         const wrapper = shallow(<div>{column.render(tokenWithNoKey)}</div>);
         expect(wrapper.text()).toBe('');
       });
+
+      it('renders an EuiCopy component with the key', () => {
+        const column = columns[2];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        expect(wrapper.find(EuiCopy).props().textToCopy).toEqual('abc-123');
+      });
+
+      it('renders a HiddenText component with the key', () => {
+        const column = columns[2];
+        const wrapper = shallow(<div>{column.render(token)}</div>)
+          .find(EuiCopy)
+          .dive();
+        expect(wrapper.find(HiddenText).props().text).toEqual('abc-123');
+      });
+
+      it('renders a Key component', () => {
+        const column = columns[2];
+        const wrapper = shallow(<div>{column.render(token)}</div>)
+          .find(EuiCopy)
+          .dive()
+          .find(HiddenText)
+          .dive();
```

Comment:
Oof haha, the pains of shallow() and render props. Appreciate the thoroughness here!

Reply:
This was definitely painful.